{"name": "All key widgets and UI elements are visible on dashboard", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.StaleElementReferenceException: Message: stale element reference: stale element not found in the current frame\n  (Session info: chrome=138.0.7204.49); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#stale-element-reference-exception\nStacktrace:\n0   chromedriver                        0x0000000104d42654 cxxbridge1$str$ptr + 2723108\n1   chromedriver                        0x0000000104d3a8c8 cxxbridge1$str$ptr + 2690968\n2   chromedriver                        0x000000010488e714 cxxbridge1$string$len + 90428\n3   chromedriver                        0x00000001048943c8 cxxbridge1$string$len + 114160\n4   chromedriver                        0x0000000104896710 cxxbridge1$string$len + 123192\n5   chromedriver                        0x00000001048967b8 cxxbridge1$string$len + 123360\n6   chromedriver                        0x00000001048d8328 cxxbridge1$string$len + 392528\n7   chromedriver                        0x0000000104918784 cxxbridge1$string$len + 655788\n8   chromedriver                        0x0000000104916de8 cxxbridge1$string$len + 649232\n9   chromedriver                        0x00000001048c99c8 cxxbridge1$string$len + 332784\n10  chromedriver                        0x0000000104d06278 cxxbridge1$str$ptr + 2476360\n11  chromedriver                        0x0000000104d0950c cxxbridge1$str$ptr + 2489308\n12  chromedriver                        0x0000000104ce7a64 cxxbridge1$str$ptr + 2351412\n13  chromedriver                        0x0000000104d09d94 cxxbridge1$str$ptr + 2491492\n14  chromedriver                        0x0000000104cd8d58 cxxbridge1$str$ptr + 2290728\n15  chromedriver                        0x0000000104d29d60 cxxbridge1$str$ptr + 2622512\n16  chromedriver                        0x0000000104d29eec cxxbridge1$str$ptr + 2622908\n17  chromedriver                        0x0000000104d3a514 cxxbridge1$str$ptr + 2690020\n18  libsystem_pthread.dylib             0x000000018ec0df94 _pthread_start + 136\n19  libsystem_pthread.dylib             0x000000018ec08d34 thread_start + 8", "trace": "self = <tests.test_dashboard.TestDashBoard object at 0x11018b8e0>\ndriver = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"2df22097ecba7b652a85c18ca71c1878\")>\nconfig = {'browser': 'chrome', 'client': 'YATRA', 'env': 'QA', 'execution': 'local', ...}\n\n    @allure.title(\"All key widgets and UI elements are visible on dashboard\")\n    @pytest.mark.smoke\n    def test_dashboard_widgets_visible(self, driver, config):\n        login_page = LoginPage(driver)\n        dashboard_page = login_page.login(config[\"username\"], config[\"password\"])\n>       assert dashboard_page.is_widget_visible(\"piechart\"), \"Charts widget is not visible\"\n\ntest_dashboard.py:38: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n../pages/dashboard_page.py:42: in is_widget_visible\n    self.switch_to_dashboard_iframe()\n../pages/dashboard_page.py:35: in switch_to_dashboard_iframe\n    self.driver.switch_to.frame(iframe)\n../../.venv/lib/python3.9/site-packages/selenium/webdriver/remote/switch_to.py:95: in frame\n    self._driver.execute(Command.SWITCH_TO_FRAME, {\"id\": frame_reference})\n../../.venv/lib/python3.9/site-packages/selenium/webdriver/remote/webdriver.py:448: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x11018bdc0>\nresponse = {'status': 404, 'value': '{\"value\":{\"error\":\"stale element reference\",\"message\":\"stale element reference: stale elemen...00018ec0df94 _pthread_start + 136\\\\n19  libsystem_pthread.dylib             0x000000018ec08d34 thread_start + 8\\\\n\"}}'}\n\n    def check_response(self, response: Dict[str, Any]) -> None:\n        \"\"\"Checks that a JSON response from the WebDriver does not have an\n        error.\n    \n        :Args:\n         - response - The JSON response from the WebDriver server as a dictionary\n           object.\n    \n        :Raises: If the response contains an error message.\n        \"\"\"\n        status = response.get(\"status\", None)\n        if not status or status == ErrorCode.SUCCESS:\n            return\n        value = None\n        message = response.get(\"message\", \"\")\n        screen: str = response.get(\"screen\", \"\")\n        stacktrace = None\n        if isinstance(status, int):\n            value_json = response.get(\"value\", None)\n            if value_json and isinstance(value_json, str):\n                import json\n    \n                try:\n                    value = json.loads(value_json)\n                    if len(value) == 1:\n                        value = value[\"value\"]\n                    status = value.get(\"error\", None)\n                    if not status:\n                        status = value.get(\"status\", ErrorCode.UNKNOWN_ERROR)\n                        message = value.get(\"value\") or value.get(\"message\")\n                        if not isinstance(message, str):\n                            value = message\n                            message = message.get(\"message\")\n                    else:\n                        message = value.get(\"message\", None)\n                except ValueError:\n                    pass\n    \n        exception_class: Type[WebDriverException]\n        e = ErrorCode()\n        error_codes = [item for item in dir(e) if not item.startswith(\"__\")]\n        for error_code in error_codes:\n            error_info = getattr(ErrorCode, error_code)\n            if isinstance(error_info, list) and status in error_info:\n                exception_class = getattr(ExceptionMapping, error_code, WebDriverException)\n                break\n        else:\n            exception_class = WebDriverException\n    \n        if not value:\n            value = response[\"value\"]\n        if isinstance(value, str):\n            raise exception_class(value)\n        if message == \"\" and \"message\" in value:\n            message = value[\"message\"]\n    \n        screen = None  # type: ignore[assignment]\n        if \"screen\" in value:\n            screen = value[\"screen\"]\n    \n        stacktrace = None\n        st_value = value.get(\"stackTrace\") or value.get(\"stacktrace\")\n        if st_value:\n            if isinstance(st_value, str):\n                stacktrace = st_value.split(\"\\n\")\n            else:\n                stacktrace = []\n                try:\n                    for frame in st_value:\n                        line = frame.get(\"lineNumber\", \"\")\n                        file = frame.get(\"fileName\", \"<anonymous>\")\n                        if line:\n                            file = f\"{file}:{line}\"\n                        meth = frame.get(\"methodName\", \"<anonymous>\")\n                        if \"className\" in frame:\n                            meth = f\"{frame['className']}.{meth}\"\n                        msg = \"    at %s (%s)\"\n                        msg = msg % (meth, file)\n                        stacktrace.append(msg)\n                except TypeError:\n                    pass\n        if exception_class == UnexpectedAlertPresentException:\n            alert_text = None\n            if \"data\" in value:\n                alert_text = value[\"data\"].get(\"text\")\n            elif \"alert\" in value:\n                alert_text = value[\"alert\"].get(\"text\")\n            raise exception_class(message, screen, stacktrace, alert_text)  # type: ignore[call-arg]  # mypy is not smart enough here\n>       raise exception_class(message, screen, stacktrace)\nE       selenium.common.exceptions.StaleElementReferenceException: Message: stale element reference: stale element not found in the current frame\nE         (Session info: chrome=138.0.7204.49); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#stale-element-reference-exception\nE       Stacktrace:\nE       0   chromedriver                        0x0000000104d42654 cxxbridge1$str$ptr + 2723108\nE       1   chromedriver                        0x0000000104d3a8c8 cxxbridge1$str$ptr + 2690968\nE       2   chromedriver                        0x000000010488e714 cxxbridge1$string$len + 90428\nE       3   chromedriver                        0x00000001048943c8 cxxbridge1$string$len + 114160\nE       4   chromedriver                        0x0000000104896710 cxxbridge1$string$len + 123192\nE       5   chromedriver                        0x00000001048967b8 cxxbridge1$string$len + 123360\nE       6   chromedriver                        0x00000001048d8328 cxxbridge1$string$len + 392528\nE       7   chromedriver                        0x0000000104918784 cxxbridge1$string$len + 655788\nE       8   chromedriver                        0x0000000104916de8 cxxbridge1$string$len + 649232\nE       9   chromedriver                        0x00000001048c99c8 cxxbridge1$string$len + 332784\nE       10  chromedriver                        0x0000000104d06278 cxxbridge1$str$ptr + 2476360\nE       11  chromedriver                        0x0000000104d0950c cxxbridge1$str$ptr + 2489308\nE       12  chromedriver                        0x0000000104ce7a64 cxxbridge1$str$ptr + 2351412\nE       13  chromedriver                        0x0000000104d09d94 cxxbridge1$str$ptr + 2491492\nE       14  chromedriver                        0x0000000104cd8d58 cxxbridge1$str$ptr + 2290728\nE       15  chromedriver                        0x0000000104d29d60 cxxbridge1$str$ptr + 2622512\nE       16  chromedriver                        0x0000000104d29eec cxxbridge1$str$ptr + 2622908\nE       17  chromedriver                        0x0000000104d3a514 cxxbridge1$str$ptr + 2690020\nE       18  libsystem_pthread.dylib             0x000000018ec0df94 _pthread_start + 136\nE       19  libsystem_pthread.dylib             0x000000018ec08d34 thread_start + 8\n\n../../.venv/lib/python3.9/site-packages/selenium/webdriver/remote/errorhandler.py:232: StaleElementReferenceException"}, "steps": [{"name": "Enter email and password and click Sign In.", "status": "passed", "parameters": [{"name": "username", "value": "'sumit@kgrp.in'"}, {"name": "password", "value": "'HRzyUJi0'"}], "start": 1751352929405, "stop": 1751352929919}], "attachments": [{"name": "test_dashboard_widgets_visible_2025-07-01_12-25-38.png", "source": "073174fc-a4ff-4a3d-b1af-e83a1038a480-attachment.png", "type": "image/png"}, {"name": "log", "source": "df3c6c7d-ea72-4cf0-9d08-a63131a62ee1-attachment.txt", "type": "text/plain"}, {"name": "stdout", "source": "68e32c86-a158-4c7b-967f-048d8945e96b-attachment.txt", "type": "text/plain"}], "start": 1751352929405, "stop": 1751352938028, "uuid": "8077d419-36f7-4ae2-bdf0-17357234ab2f", "historyId": "2239709bfbdd072b434c77afa582d375", "testCaseId": "2239709bfbdd072b434c77afa582d375", "fullName": "src.tests.test_dashboard.TestDashBoard#test_dashboard_widgets_visible", "labels": [{"name": "tag", "value": "smoke"}, {"name": "parentSuite", "value": "src.tests"}, {"name": "suite", "value": "test_dashboard"}, {"name": "subSuite", "value": "TestDashBoard"}, {"name": "host", "value": "macs-MacBook-Pro.local"}, {"name": "thread", "value": "65498-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "src.tests.test_dashboard"}]}